typescript-requiring-type-checking.js

```diff
@@ -15,23 +15,23 @@ module.exports = {
         /*
             TypeScript rules outside of the recommended configuration
         */
-        // '@typescript-eslint/naming-convention': 'error',
-        // '@typescript-eslint/no-base-to-string': 'error',
-        // '@typescript-eslint/no-confusing-void-expression': 'error',
-        // '@typescript-eslint/no-unnecessary-boolean-literal-compare': 'error',
-        // '@typescript-eslint/no-unnecessary-condition': 'error',
-        // '@typescript-eslint/no-unnecessary-qualifier': 'error',
-        // '@typescript-eslint/no-unnecessary-type-arguments': 'error',
-        // '@typescript-eslint/non-nullable-type-assertion-style': 'error',
-        // '@typescript-eslint/prefer-includes': 'error',
-        // '@typescript-eslint/prefer-nullish-coalescing': 'error',
-        // '@typescript-eslint/prefer-readonly': 'error',
-        // '@typescript-eslint/prefer-readonly-parameter-types': 'error',
-        // '@typescript-eslint/prefer-reduce-type-parameter': 'error',
-        // '@typescript-eslint/prefer-string-starts-ends-with': 'error',
-        // '@typescript-eslint/promise-function-async': 'error',
-        // '@typescript-eslint/require-array-sort-compare': 'error',
-        // '@typescript-eslint/strict-boolean-expressions': 'error',
-        // '@typescript-eslint/switch-exhaustiveness-check': 'error'
+        '@typescript-eslint/naming-convention': 'error',
+        '@typescript-eslint/no-base-to-string': 'error',
+        '@typescript-eslint/no-confusing-void-expression': 'error',
+        '@typescript-eslint/no-unnecessary-boolean-literal-compare': 'error',
```

Comment:
[@typescript-eslint/no-unnecessary-boolean-literal-compare](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-boolean-literal-compare.md)
"Warning: Do not use this rule when strictNullChecks is disabled. ESLint is not able to distinguish between false and undefined or null values. This can cause unintended code changes when using autofix."